# Exploratory data analysis on London Data

This notebook will serve for exploring the London Dataset and gainging insight on how to process it 

First, we recuperate the necessary packages 

In [1]:
import pandas as pd
import numpy as np

The data is divided in 163 datasets of 1 million entries.

We upload the data from several csv file and merge them together

In [2]:
df = pd.read_csv(r"..\Data\Londondata\Power-Networks-LCL-June2015(withAcornGps)v2_34.csv")

for i in range(1,3): 
    dataframe = pd.read_csv(r"..\Data\Londondata\Power-Networks-LCL-June2015(withAcornGps)v2_"+str(i)+".csv")
    df = df.append(dataframe)


In [6]:
df.head()

LCLid stdorToU                     DateTime KWH/hh (per half hour)   \
0  MAC001262      Std  2012-08-05 04:00:00.0000000                  0.047    
1  MAC001262      Std  2012-08-05 04:30:00.0000000                  0.073    
2  MAC001262      Std  2012-08-05 05:00:00.0000000                   0.04    
3  MAC001262      Std  2012-08-05 05:30:00.0000000                  0.054    
4  MAC001262      Std  2012-08-05 06:00:00.0000000                  0.058    

     Acorn Acorn_grouped  
0  ACORN-F   Comfortable  
1  ACORN-F   Comfortable  
2  ACORN-F   Comfortable  
3  ACORN-F   Comfortable  
4  ACORN-F   Comfortable

The data ranges from 2012 to 2014, we will keep the data from 2013

In [7]:
#Get the year of observation
df["year"] = pd.to_datetime(df["DateTime"]).dt.year
df.index = df.index - 3840


#Isolate the year 2013
df = df[df["year"] == 2013]

Lets look at how many data points there is for each households (LCLid) for the 2013 year

In [8]:
df.groupby("LCLid")['KWH/hh (per half hour) '].apply(lambda x: len(x))

LCLid
MAC000002    17532
MAC000003    17530
MAC000004    17531
MAC000006    17532
MAC000007    17529
MAC000008    14505
MAC000009    17523
MAC000010    17532
MAC000011    17531
MAC000012    17531
MAC000013    17531
MAC000016      721
MAC000018    17532
MAC000019    17532
MAC000020    17530
MAC000021    17531
MAC000022    17531
MAC000023    17531
MAC000024    17531
MAC000025    17526
MAC000026    17519
MAC000027    17532
MAC000028    13401
MAC000029    17520
MAC000030    17523
MAC000032    17531
MAC000033    17531
MAC000034    17528
MAC000035    17530
MAC000036    17531
             ...  
MAC001267    17523
MAC001268    17527
MAC001269    17530
MAC001271    17531
MAC001272    17529
MAC001273    17523
MAC001275    17531
MAC001276    17526
MAC001277    17530
MAC001280    17531
MAC001282    17527
MAC001283    17532
MAC001284     2018
MAC001286    17530
MAC001287     2736
MAC001288    15658
MAC001289    17530
MAC001291    17531
MAC001292    17531
MAC001293    17531
MAC001294    17532
MAC001

We can see that most Households have around 17530 entries, which makes sens as there are 17520 half hours in a year

Some are however far from this value. As we have plenty of data, lets delete all Household with less than 17520 data points


In [13]:
groups = df.groupby("LCLid")['KWH/hh (per half hour) '].apply(lambda x: len(x))
groups = groups[groups>17520]

ind = groups.index

df = df[df["LCLid"].isin(ind)]